pkg/upgraders/postupgradestep.go

```diff
@@ -0,0 +1,111 @@
+package upgraders
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/openshift/managed-upgrade-operator/config"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	fioNamespace    string = "openshift-file-integrity"
+	fioObject       string = "osd-fileintegrity"
+	frOCMBaseDomain string = "openshiftusgov.com"
+)
+
+var reinitAnnotation = map[string]string{"file-integrity.openshift.io/re-init": ""}
+
+// ugConfigManager stores the configManager section of the upgradeconfig manager configmap
+type ugConfigManager struct {
+	Source     string `yaml:"source"`
+	OcmBaseURL string `yaml:"ocmBaseUrl"`
+}
+
+// ugConfigManagerSpec stores the config.yaml section of the upgradeconfig manager configmap
+type ugConfigManagerSpec struct {
+	Config ugConfigManager `yaml:"configManager"`
+}
+
+// PostUpgradeProcedures are any misc tasks that are needed to be completed after an upgrade has finished to ensure healthy state
+// Currently the only task is to reinit file integrity operator due to changes that come from upgrades
+func (c *clusterUpgrader) PostUpgradeProcedures(ctx context.Context, logger logr.Logger) (bool, error) {
+
+	frCluster, err := c.frClusterCheck(ctx)
+	if err != nil {
+		return false, err
+	}
+	if !frCluster {
+		logger.Info("Non-FedRAMP environment...skipping PostUpgradeFIOReInit ")
+		return true, nil
+	}
+	err = c.postUpgradeFIOReInit(ctx, logger)
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// frClusterCheck checks to see if the upgrading cluster is a FedRAMP cluster to determine if we need to re-init the File Integrity Operator
+func (c *clusterUpgrader) frClusterCheck(ctx context.Context) (bool, error) {
+	ocmConfig := &corev1.ConfigMap{}
+	err := c.client.Get(context.TODO(), client.ObjectKey{Namespace: config.OperatorNamespace, Name: config.ConfigMapName}, ocmConfig)
+
+	if err != nil {
+		return false, fmt.Errorf("failed to fetch %s config map to parse: %v", config.ConfigMapName, err)
+	}
+
+	var cm ugConfigManagerSpec
+	data := fmt.Sprint(ocmConfig.Data["config.yaml"])
+	err = yaml.Unmarshal([]byte(data), &cm)
+
+	if err != nil {
+		return false, fmt.Errorf("failed to parse %s config map for OCM URL: %v", config.ConfigMapName, err)
+	}
```

Comment:
Rather than do yaml wrangling in here, it would be possible to add in something like a `isFedRamp` function into this, which is the part of the code built around extracting upgrade-related config from the ConfigMap:

https://github.com/openshift/managed-upgrade-operator/blob/master/pkg/upgraders/config.go

I am additionally wondering if there are ways we could identify Fedramp clusters that don't involve the combination of the hardcoded domain name and the OCM type. Maybe we could introduce a boolean new field to the `ConfigMap` which identifies FedRamp clusters? And then distribute a Fedramp-specific version of the ConfigMap? Spitballing ideas here.

Reply:
I think I like the idea of a FedRAMP boolean in the upgrade configmap. I also think W. Trevor King has a good point here that perhaps using something specific to the FileIntegrity object would be simpler than grabbing the upgrade config, but we'd still be looking at a hard coded value perhaps. We could very easily add a label to the file integrity we create in the [OLM template](https://gitlab.cee.redhat.com/service/file-integrity-operator-catalog/-/blob/master/hack/olm-registry/olm-artifacts-template.yaml#L129) since this is how we're deploying everything into FedRAMP clusters. At that point we can check if the file integrity even exists and if it does check the label and go from there. Not sure which is better?

Reply:
@mrbarge I think I made some of the changes you suggested to the config.go file, if I understood right. I doubt this is working yet as I'm not sure where the config value would be consumed and if it is at the moment, but wanted to make sure i was grokking what you meant. This logic would assume that the muo-config in cluster would look something like
```
data:
  config.yaml: |
   environment:
     fedramp: true
    upgradeType: OSD
    configManager:
      source: OCM
      ocmBaseUrl: ${OCM_BASE_URL}
```
This way when the upgraderConfig is created/parsed we can greatly simplify the check later in post upgrade like mentioned
```
func (c *clusterUpgrader) PostUpgradeProcedures(ctx context.Context, logger logr.Logger) (bool, error) {

	if !c.config.Environment.IsFedramp() {
		logger.Info("Non-FedRAMP environment...skipping PostUpgradeFIOReInit ")
		return true, nil
	}
	err := c.postUpgradeFIOReInit(ctx, logger)
	if err != nil {
		return false, err
	}
	return true, nil
}
```

Does this seem like a more favorable solution that I can work on getting added to submit?

Reply:
@mrbarge updates seem to work just fine using this method! This is much nicer. I've added the logs for my test in commercial to the [jira card](https://issues.redhat.com/browse/OSD-9594?focusedCommentId=20850505&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-20850505) using these changes to show that it works. One test with the `managed-upgrade-operator-confg` having fedramp set to false, the other true.

Reply:
@tonytheleg thanks for making this change, it's exactly what I was thinking of. Looks good and I gave it a quick test successfully as well.
/lgtm